cxx/BUILD

```diff
@@ -60,6 +60,35 @@ cc_binary(
     ],
 )
 
+cc_library(
+    name = "noisy_relation",
+    hdrs = ["noisy_relation.hh"],
+    visibility = [":__subpackages__"],
+    deps = [
+        ":domain",
+        ":non_noisy_relation",
+        ":relation",
+        ":util_distribution_variant",
+        ":util_hash",
+        ":util_math",
+        "//distributions:base"
+    ],
+)
+
+cc_library(
+    name = "non_noisy_relation",
```

Comment:
Maybe clean_relation instead of non_noisy (here and elsewhere)?

Reply:
Done (I was a little torn because I think "non noisy" is more self-documenting, but I like that "clean" is concise and easier to visually disambiguate from "noisy").